src/sentry/sentry_metrics/querying/data_v2/transformation.py

```diff
@@ -103,24 +104,27 @@ def _assert_transformation_preconditions(self) -> tuple[datetime, datetime, int]
 
     def _build_intermediate_results(
         self,
-    ) -> tuple[OrderedDict[GroupKey, OrderedDict[str, GroupValue]], list[QueryMeta]]:
+    ) -> tuple[list[OrderedDict[GroupKey, GroupValue]], list[list[QueryMeta]]]:
         """
         Builds a tuple of intermediate groups and metadata which is used to efficiently transform the query results.
         """
-        intermediate_groups: OrderedDict[GroupKey, OrderedDict[str, GroupValue]] = OrderedDict()
-        intermediate_meta: list[QueryMeta] = []
-
-        def _add_to_intermediate_groups(values, block):
-            for value in values:
-                # We compute a list containing all the group values.
+        queries_groups: list[OrderedDict[GroupKey, GroupValue]] = []
+        queries_meta: list[list[QueryMeta]] = []
+
+        def _add_to_query_groups(
+            rows: Sequence[Mapping[str, Any]],
+            group_bys: list[str],
+            query_groups: OrderedDict[GroupKey, GroupValue],
+            block: Callable[[Mapping[str, Any], GroupValue], None],
```

Comment:
nit: took me a while to figure out this doesnt have anything to do with metric blocking. This is just an accumulator fn, right? Maybe we should find a better name for it.